waypoints.proto

```diff
@@ -0,0 +1,367 @@
+syntax = "proto3";
+
+option java_package = "com.geeksville.mesh";
+option java_outer_classname = "WaypointType";
+option optimize_for = LITE_RUNTIME;
+option go_package = "github.com/meshtastic/go/generated";
+
+/*
+ * Goal of this type is to provide general categories for Waypoints.
+ * This way an external interface can show them as a special icons.
+ * Therefore maps with waypoints will be a lot easier to understand.
+ * Focus for categories is on navigation, organisation of equipment and special places to know about.
+ *
+ * Categories with 128 entries each:
+ * (The first value in each category is always generic.)
+ *
+ * UNKNOWN:        0
+ * GENERAL-POINTS: 1    - 127
+ * LAND-MARKS:     128  - 255
+ * BUILDINGS:      256  - 383
+ * PLACES:         384  - 511
+ * VEHICLES:       512  - 639
+ * RESSOURCES:     640  - 767
+ */
+enum WaypointType {
```

Comment:
The size of this enum raises a concern with me. It'll take up at least 16 bits over the air. Please consider rescaling the enum where 127 is the last possible value. That'll limit the expense of this to 8 bits.